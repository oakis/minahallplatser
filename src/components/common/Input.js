import React from 'react';
import {TextInput, View} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {Text, Spinner} from '@common';
import {colors, component, metrics} from '@style';

export const Input = ({
  style,
  value,
  label,
  icon,
  iconRight = null,
  iconRightPress = null,
  iconSize = 24,
  placeholder,
  onChangeText,
  autoFocus = false,
  returnKeyType,
  keyboardType,
  secureTextEntry = false,
  loading = false,
  underlineColorAndroid = colors.primary,
  onFocus,
  multiline = false,
}) => {
  return (
    <View style={[component.input.container, style]}>
      {label ? <Text>{label}</Text> : null}
      {icon ? (
        <Icon
          name={icon}
          size={iconSize}
          style={{
            marginLeft: metrics.margin.sm,
            marginRight: metrics.margin.sm,
          }}
        />
      ) : null}
      <TextInput
        style={{marginTop: 0, flex: 1}}
        placeholder={placeholder}
        onChangeText={onChangeText}
        autoFocus={autoFocus}
        returnKeyType={returnKeyType}
        value={value}
        underlineColorAndroid={underlineColorAndroid}
        keyboardType={keyboardType}
        secureTextEntry={secureTextEntry}
        autoCorrect={false}
        autoCapitalize="none"
        onFocus={onFocus}
        multiline={multiline}
      />
      {iconRight !== null ? (
        <View style={{marginLeft: metrics.margin.sm}}>
          {loading ? (
            <Spinner
              color={colors.primary}
              style={{
                padding: metrics.padding.md,
                paddingRight: metrics.padding.md + 4,
              }}
            />
          ) : (
            <Icon
              onPress={iconRightPress}
              name={iconRight}
              size={iconSize}
              style={{padding: metrics.padding.md}}
            />
          )}
        </View>
      ) : null}
    </View>
  );
};
