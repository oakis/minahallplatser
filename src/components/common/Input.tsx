import React from 'react';
import {TextInput, View} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {Text, Spinner} from '@common';
import {colors, component, metrics} from '@style';

type InputProps = {
  style: object;
  value: string;
  label: string;
  icon: string | null;
  iconRight: string | null;
  iconRightPress: () => void;
  iconSize: number;
  placeholder: string;
  onChangeText: () => void;
  autoFocus: boolean;
  secureTextEntry: boolean;
  loading: boolean;
  underlineColorAndroid: string;
  onFocus: () => void;
  multiline: boolean;
};

export const Input = ({
  style,
  value,
  label,
  icon = null,
  iconRight = null,
  iconRightPress = () => {},
  iconSize = 24,
  placeholder,
  onChangeText,
  autoFocus = false,
  secureTextEntry = false,
  loading = false,
  underlineColorAndroid = colors.primary,
  onFocus,
  multiline = false,
}: InputProps): JSX.Element => {
  return (
    <View style={[component.input.container, style]}>
      {label ? <Text>{label}</Text> : null}
      {icon ? (
        <Icon
          name={icon}
          size={iconSize}
          style={{
            marginLeft: metrics.margin.sm,
            marginRight: metrics.margin.sm,
          }}
        />
      ) : null}
      <TextInput
        style={{marginTop: 0, flex: 1}}
        placeholder={placeholder}
        onChangeText={onChangeText}
        autoFocus={autoFocus}
        value={value}
        underlineColorAndroid={underlineColorAndroid}
        secureTextEntry={secureTextEntry}
        autoCorrect={false}
        autoCapitalize="none"
        onFocus={onFocus}
        multiline={multiline}
      />
      {iconRight !== null ? (
        <View style={{marginLeft: metrics.margin.sm}}>
          {loading ? (
            <Spinner
              color={colors.primary}
              style={{
                padding: metrics.padding.md,
                paddingRight: metrics.padding.md + 4,
              }}
            />
          ) : (
            <Icon
              onPress={iconRightPress}
              name={iconRight}
              size={iconSize}
              style={{padding: metrics.padding.md}}
            />
          )}
        </View>
      ) : null}
    </View>
  );
};
