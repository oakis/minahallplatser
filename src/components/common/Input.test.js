import React from 'react';
import {Input} from './Input';
import {render} from '@testing-library/react-native';

it('should match snapshot', () => {
  const {toJSON} = render(
    <Input label="text" icon="mail" iconRight="vpn-key" />,
  );
  expect(toJSON()).toMatchSnapshot();
});
