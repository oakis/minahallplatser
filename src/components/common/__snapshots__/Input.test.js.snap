// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 10,
        "shadowColor": "#282828",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 5,
      },
      undefined,
    ]
  }
>
  <DefaultFont>
    text
  </DefaultFont>
  <MaterialIcon
    name="mail"
    size={24}
    style={
      Object {
        "marginLeft": 5,
        "marginRight": 5,
      }
    }
  />
  <TextInput
    autoCapitalize="none"
    autoCorrect={false}
    autoFocus={false}
    multiline={false}
    secureTextEntry={false}
    style={
      Object {
        "flex": 1,
        "marginTop": 0,
      }
    }
    underlineColorAndroid="#779ECB"
  />
  <View
    style={
      Object {
        "marginLeft": 5,
      }
    }
  >
    <MaterialIcon
      name="vpn-key"
      onPress={null}
      size={24}
      style={
        Object {
          "padding": 10,
        }
      }
    />
  </View>
</View>
`;
